packages/spacecat-shared-http-utils/src/auth/auth-wrapper.js

```diff
@@ -48,6 +53,44 @@ export function authWrapper(fn, opts = {}) {
       return new Response('Unauthorized', { status: 401 });
     }
 
+    if (!context.auth) {
+      context.auth = {
+        hasScopes: async (scopes) => {
+          // Pull the api-key from x-api-key header
+          const apiKeyFromHeader = context.pathInfo?.headers['x-api-key'];
+
+          if (!hasText(apiKeyFromHeader)) {
+            return new Response('Unauthorized', { status: 401 });
```

Comment:
I would like to see this function always return a boolean, such that the individual controllers can decide how to handle a falsey result. 

Logging the reason for the `false` should happen, so we can diagnose issues on our side. 

On the other hand, we could consider returning a small object that would have details on the reason that hasScopes was false, ie:

```
{
  result: true
}

// or, in the falsey case
{
  result: false,
  reason: 'API key has been revoked'
}

// or, in the case of a missing scope (organization.read_all)
{
  result: false,
  reason: 'API key is missing the ["organization.read_all"] scope(s) required for this resource'
}
```

What do you think?